mlflow/tracking/_tracking_service/utils.py

```diff
@@ -171,10 +170,33 @@ def _get_databricks_rest_store(store_uri, **_):
     return RestStore(partial(get_databricks_host_creds, store_uri))
 
 
+def _get_databricks_uc_rest_store(store_uri, **_):
+    from mlflow.exceptions import MlflowException
+    from mlflow.version import VERSION
+
+    global _tracking_store_registry
+    raise MlflowException(
+        f"Detected Unity Catalog tracking URI '{store_uri}'. "
+        f"Setting the tracking URI to a Unity Catalog backend is not supported in the current "
+        f"version of the MLflow client ({VERSION}). "
+        f"Please specify a different tracking URI via mlflow.set_tracking_uri, with "
+        f"one of the supported schemes: "
+        f"{list(_tracking_store_registry._registry.keys())}. "
```

Comment:
The error message contains `'databricks-uc'`. Should we remove it if it's unsupported?

---

mlflow.exceptions.MlflowException: Detected Unity Catalog tracking URI 'databricks-uc'. Setting the tracking URI to a Unity Catalog backend is currently unsupported. Please specify a different tracking URI via mlflow.set_tracking_uri, with one of the following supported schemes: ['', 'file', 'databricks', 👉 'databricks-uc', 'http', 'https', 'postgresql', 'mysql', 'sqlite', 'mssql', 'file-plugin']. If you're trying to access models in the Unity Catalog, please upgrade to the latest version of the MLflow Python client, then specify a Unity Catalog model registry URI via mlflow.set_registry_uri('databricks-uc') or mlflow.set_registry_uri('databricks-uc://profile_name'), where 'profile_name' is the name of the Databricks CLI profile to use for authentication. Be sure to leave the tracking URI configured to use one of the supported schemes listed above.

Reply:
Nice catch! Yes, good call

Reply:
Verified after the latest commit that the message doesn't suggest "databricks-uc" as a supported scheme:

```
~/mlflow reserve-data…g-uri-scheme ❯ python                                                                                      Py base Py 3.8.13 09:01:08 PM
Python 3.8.13 (default, Jan 25 2023, 19:47:36) 
[Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import mlflow; mlflow.set_tracking_uri("databricks-uc"); mlflow.log_param("a", "b")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sid.murching/mlflow/mlflow/tracking/fluent.py", line 544, in log_param
    run_id = _get_or_start_run().info.run_id
  File "/Users/sid.murching/mlflow/mlflow/tracking/fluent.py", line 1552, in _get_or_start_run
    return start_run()
  File "/Users/sid.murching/mlflow/mlflow/tracking/fluent.py", line 278, in start_run
    client = MlflowClient()
  File "/Users/sid.murching/mlflow/mlflow/tracking/client.py", line 69, in __init__
    self._tracking_client = TrackingServiceClient(final_tracking_uri)
  File "/Users/sid.murching/mlflow/mlflow/tracking/_tracking_service/client.py", line 51, in __init__
    self.store
  File "/Users/sid.murching/mlflow/mlflow/tracking/_tracking_service/client.py", line 55, in store
    return utils._get_store(self.tracking_uri)
  File "/Users/sid.murching/mlflow/mlflow/tracking/_tracking_service/utils.py", line 218, in _get_store
    return _tracking_store_registry.get_store(store_uri, artifact_uri)
  File "/Users/sid.murching/mlflow/mlflow/tracking/_tracking_service/registry.py", line 39, in get_store
    return self._get_store_with_resolved_uri(resolved_store_uri, artifact_uri)
  File "/Users/sid.murching/mlflow/mlflow/tracking/_tracking_service/registry.py", line 49, in _get_store_with_resolved_uri
    return builder(store_uri=resolved_store_uri, artifact_uri=artifact_uri)
  File "/Users/sid.murching/mlflow/mlflow/tracking/_tracking_service/utils.py", line 185, in _get_databricks_uc_rest_store
    raise MlflowException(
mlflow.exceptions.MlflowException: Detected Unity Catalog tracking URI 'databricks-uc'. Setting the tracking URI to a Unity Catalog backend is not supported in the current version of the MLflow client (2.1.2.dev0). Please specify a different tracking URI via mlflow.set_tracking_uri, with one of the supported schemes: ['', 'file', 'databricks', 'http', 'https', 'postgresql', 'mysql', 'sqlite', 'mssql', 'file-plugin']. If you're trying to access models in the Unity Catalog, please upgrade to the latest version of the MLflow Python client, then specify a Unity Catalog model registry URI via mlflow.set_registry_uri('databricks-uc') or mlflow.set_registry_uri('databricks-uc://profile_name'), where 'profile_name' is the name of the Databricks CLI profile to use for authentication. Be sure to leave the tracking URI configured to use one of the supported schemes listed above.
```